packages/session-replay/src/BacktraceSessionRecorder.ts

```diff
@@ -0,0 +1,80 @@
+import { BacktraceAttachment } from '@backtrace/sdk-core';
+import { eventWithTime } from '@rrweb/types';
+import { record } from 'rrweb';
+import { BacktraceSessionRecorderOptions } from './options';
+
+function defaultIfNotFalse<T>(value: T | false, defaultValue?: T) {
+    if (value === false) {
+        return undefined;
+    }
+
+    return value ?? defaultValue;
+}
+
+export class BacktraceSessionRecorder implements BacktraceAttachment {
+    public readonly name = 'bt-session-replay-0';
+    public readonly type = 'dynamic';
+
+    private readonly _maxEventCount?: number;
+    private readonly _maxTime?: number;
+
+    private _previousEvents?: eventWithTime[] = [];
+    private _events?: eventWithTime[] = [];
+
+    private _stop?: () => void;
+
+    constructor(private readonly _options: BacktraceSessionRecorderOptions) {
+        this._events = [];
+        this._maxEventCount = defaultIfNotFalse(_options.maxEventCount, 100);
```

Comment:
The user can disable a limit and keep it from setting to default. For example, if they want to use only `maxTime`, they would want to set `maxEventCount` to `false` to prevent it being set by default to 100.

Reply:
Makes sense. Makes me wonder if it might be a confusing pattern for the user. Would it make more sense to just have a separate field for disabling maxEventCount and maxTime that would default to being enabled and take precedence if disabled over any max set?

Reply:
Agree with Adam, it's confusing. As the user I wouldn't expect maxEventCount can accept false

Reply:
I added new options to disable the limits.